Upgrade angular V7.0.3 and enhance

yd021976 opened the pull request:
Hi,

I submit some changes :

- Upgrade workspace to angular V7.0.0 compliant (use of library project)
- Some enhancements :
1. Provide "permission states" in custom strategy callback
2. Provide "permissions states" in event emitters (authorized and unauthorized)

AlexKhymenko commented:
@yd021976  Can You please resolve conflicts so I can look into it :-) But i don't have much time right now. And Thank You For this funcitonality

yd021976 commented:
@AlexKhymenko Thank you, can you give me **write access** to the repository or the conflict files so I can resolve conflicts ?
<img width="707" alt="image" src="https://user-images.githubusercontent.com/2945570/48046109-6e31ce00-e193-11e8-8f61-d2448b365d25.png">

AlexKhymenko commented:
@yd021976 Actually there is a message that 'this conflicts are too complex to resolve in the web editor' Can You merge master to Your branch?

yd021976 commented:
@AlexKhymenko Hi, i've fixed conflicts. But seems that as I modified the project structure to be Angular 7 compliant, it breaks your travis CI build (actually, package.json has been moved to "./application" instead of being in ./ before ==> I think there is a lot of other changes in my PR that will break Travis CI.

yd021976 commented:
@AlexKhymenko Hi again :-) I confirm that the new project structure definitively break your Travis CI.
When you have time can you update travis CI to validate my PR ?

Thank you !

yd021976 commented:
@AlexKhymenko , I've made some changes in package.json so that Travis CI should run with success. Unfortunately, Angular V6/7 needs Node js >= v8.9

Could you please change the Travis CI config with a required node version compatible with angular >= V6 ? (i.e Node V8.9)

Thank you

AlexKhymenko commented:
@yd021976 Thank You!! Very much i will Look asap. Just have a lot of work :-(

AlexKhymenko commented:
@yd021976 Thank You Again. Just There are a lot of changes and not so much time :-(
